source/MaterialXCore/UnitConverter.cpp

```diff
@@ -0,0 +1,88 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+/**
+* Copyright 2019 Autodesk, Inc.
+* All rights reserved.
+*
+* This computer source code and related instructions and comments are the unpublished confidential
+* and proprietary information of Autodesk, Inc. and are protected under Federal copyright and state
+* trade secret law. They may not be disclosed to, copied or used by any third party without the
+* prior written consent of Autodesk, Inc.
+*
+* @file Units.cpp
+* @brief Units support
+*/
+
+
+#include <MaterialXCore/Util.h>
+#include <MaterialXCore/Value.h>
+#include <MaterialXCore/UnitConverter.h>
+
+namespace MaterialX
+{
+
+LengthUnitConverter::LengthUnitConverter(UnitTypeDefPtr unitTypeDef)
```

Comment:
This is the supplied one. It is possible to use your own of course. We can have one per unit type in the core or added by users.